src/core/infrastructure/guard.js

```diff
@@ -0,0 +1,36 @@
+import Error from  './error';
+import {isUndefined} from '../services/utility';
+
+/**
+ * If value is undefined exception will be thrown
+ * @param {any} value - Value to check
+ * @param {string} name - Argument name
+ */
+export function notUndefined(value, name){
+	if (isUndefined(value)) {
+		throw new Error('guard', name);
+	}
+}
+
+
+/**
+ * If value is null or undefined exception will be thrown
+ * @param {any} value - Value to check
+ * @param {string} name - Argument name
+ */
+export function notNull(value, name){
+	if (value === null || isUndefined(value)) {
```

Comment:
I'm not sure, we'll try to write in not nullable style, this is just infrastracture methods